fabulaws/library/wsgiautoscale/api.py

```diff
@@ -466,7 +466,14 @@ def _new(deployment, environment, role, avail_zone=None, count=1, **kwargs):
                      tags=tags, volume_size=vol_size, volume_type=vol_type,
                      deploy_user=env.deploy_user, security_groups=sec_grps,
                      **extra_args)
-        server.setup()
+        try:
+            server.setup()
+        except:
+            logger.exception('server.setup() failed. tags=%s; terminate_on_failure=%s.'
+                             '' % (tags, terminate_on_failure))
```

Comment:
If you change `% (tags, terminate_on_failure)` to `, tags, terminate_on_failire`, then the logging system will only evaluate the arguments if it actually has to (rather than doing the string interpolation before the logging system can decide).

It can be an issue when `debug` level calls have expensive-to-calculate parameters in the string interpolation (which would inadvertently slow down production even though the message doesn't get logged). That's obviously not the case here.